catalog/registry/catalog-registry-cswinputtransformer/src/main/java/ddf/catalog/registry/converter/RegistryPackageConverter.java

```diff
@@ -0,0 +1,667 @@
+import ddf.catalog.registry.common.metacard.RegistryObjectMetacardType;
+import net.opengis.cat.wrs.v_1_0_2.AnyValueType;
+import net.opengis.gml.v_3_1_1.AbstractGeometryType;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.EmailAddressType;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.InternationalStringType;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.LocalizedStringType;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.OrganizationType;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.PersonNameType;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.PersonType;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.PostalAddressType;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.RegistryObjectListType;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.RegistryObjectType;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.RegistryPackageType;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.ServiceBindingType;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.ServiceType;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.SlotType1;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.TelephoneNumberType;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.ValueListType;
+
+public class RegistryPackageConverter {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(RegistryPackageConverter.class);
+
+    private static final Map<String, String> METACARD_XML_NAME_MAP;
+
+    private static final String XML_DATE_TIME_TYPE = ":dateTime";
+
+    private static final String XML_GEO_TYPE = ":GM_Point";
+
+    private static final String REGISTRY_OBJECT_TYPE = "urn:service:catalog";
+
+    private static final String ID_PATTERN_REGEX = "urn:uuid:(.*)";
+
+    private static final Pattern ID_PATTERN = Pattern.compile(ID_PATTERN_REGEX);
+
+    static {
+        METACARD_XML_NAME_MAP = new HashMap<>();
+        METACARD_XML_NAME_MAP.put(RegistryObjectMetacardType.LIVE_DATE, "liveDate");
+        METACARD_XML_NAME_MAP.put(RegistryObjectMetacardType.DATA_START_DATE, "dataStartDate");
+        METACARD_XML_NAME_MAP.put(RegistryObjectMetacardType.DATA_END_DATE, "dataEndDate");
+        METACARD_XML_NAME_MAP.put(Metacard.MODIFIED, "lastUpdated");
+        METACARD_XML_NAME_MAP.put(RegistryObjectMetacardType.LINKS, "links");
+
+        METACARD_XML_NAME_MAP.put(Metacard.GEOGRAPHY, "location");
+        METACARD_XML_NAME_MAP.put(RegistryObjectMetacardType.REGION, "region");
+        METACARD_XML_NAME_MAP.put(RegistryObjectMetacardType.DATA_SOURCES, "inputDataSources");
+        METACARD_XML_NAME_MAP.put(RegistryObjectMetacardType.DATA_TYPES, "dataTypes");
+        METACARD_XML_NAME_MAP.put(RegistryObjectMetacardType.SECURITY_LEVEL, "securityLevel");
+        METACARD_XML_NAME_MAP.put(RegistryObjectMetacardType.SERVICE_BINDING_TYPES, "bindingType");
+        METACARD_XML_NAME_MAP.put(RegistryObjectMetacardType.SERVICE_BINDINGS, "serviceType");
+
+    }
+
+    public static Metacard getRegistryObjectMetacard(RegistryObjectType registryObject)
```

Comment:
The converter doesn't hold state and should be used as a utility class with the one public method. Added a private constructor to prevent instantiation.